tools/generator/Java.Interop.Tools.Generator.Transformation/Parser.cs

```diff
@@ -103,12 +103,12 @@ List<GenBase> ParsePackage (XElement ns, Predicate<XElement> p)
 				GenBase gen = null;
 				switch (elem.Name.LocalName) {
 				case "class":
-					if (elem.XGetAttribute ("obfuscated") == "true")
+					if (elem.XGetAttribute ("obfuscated") == "true" || elem.XGetAttribute ("visibility") == "internal")
```

Comment:
I don't understand this change and the related one for `"interface"`: why do we need to check for `./@visibility='internal'` here, but not visibility = "" (not set)/`private`/etc.?

How is non-`public` visibility handled elsewhere, and why is `internal` different?

Reply:
`private` members are not emitted by `class-parse`, so they should never exist in the `api.xml`.  (This is how we hide Kotlin internal types today: we set them to `private` in `class-parse` so they do not appear in the `api.xml`.)

`not-set` members are a little different. We still import them so they can participate in Java type resolution, and there's some logic about copying `package-private` interface methods directly to types.  However we do not emit them as managed types.

Reply:
> `not-set` members are a little different. … However we do not emit them as managed types.

How does *that* work? (The 'obvious-to-me' answer would be for e.g. `XmlApiImporter.CreateClass()` to return `null` when `Visibility==""`, but that *isn't* the case -- there are no early returns in `XmlApiImporter.CreateClass()`.)

Would it make more sense to treat `Visibility=="internal"` the same way we treat `Visibility==""`?

Not understanding how "not-set" visibility works is confusing to me, actually; `java.util.concurrent.ConcurrentHashMap.CollectionView` has no visibility (useful example) and *isn't* overridden in metadata (useful example) and *isn't* bound -- all *yay!* -- but *why* isn't it bound?  I've lost the plot, and can't readily see where/why this type isn't bound.